docs/references/strategies/languages/golang/gomodules.md

```diff
@@ -235,3 +235,24 @@ not using backoff!
 
 As a concrete step towards resolving this sort of discrepancy, we recommend running `go mod tidy` on projects regularly;
 this command should synchronize the `go.mod` file with the actual state of the project.
+
+#### Test Dependencies
+
+Sometimes the above procedure may uncover a dependency that is not reported by FOSSA but that is also not removed by `go mod tidy`.
+The other reason that a dependency may appear in `go.mod` but not in FOSSA is that it is a test dependency.
+`go.mod` itself does not label dependencies as being used only in tests, but FOSSA's Go module strategy can identify these and will exclude them.
+To verify that a direct dependency is not a test-only dependency, we recommend searching the project source code for where a package from the module is imported.
+Generally, for a declaration in `go.mod` like:
+
+```
+	github.com/prometheus/client_golang v1.12.2
+```
+
+You can search for its imports using a command like:
+
+```sh
+$ find <project_directory> -name \*.go -exec grep -Hn "github.com/prometheus/client_golang" {} \;
```

Comment:
I thought a bit about writing a tiny script to do this procedure and report back if a dep looks like a test dep. I thought it might be overkill right now, but let me know if you disagree.

Reply:
This would be immense, we've had ticket where customer had a list of dependency that they were expecting in final analysis, but all of them were test deps. Going dep by dep in debug bundle was quite time sink. Having script would greatly reduce these sort of on-call tickets.

Reply:
I agree, let's leave this simple. If it were me I would point users to `rg` so that the same advice is applicable cross-platform and is simpler, but up to you.

e.g.:
> You can search for its imports using a command like the below. Here we use [ripgrep](https://github.com/BurntSushi/ripgrep?tab=readme-ov-file#ripgrep-rg) as `rg`, but any similar tool should suffice:
> ```sh
> rg 'github.com/prometheus/client_golang' -F -l
> ```

Reply:
The cross-platform point is good. I will add a line about `rg` being a good option if `find`/`grep` aren't available. But the reason I went with a `find`/`grep` solution first is that they are generally available in the base-system of just about anything vaguely POSIX-y where `rg` is not. I'd like something a user can just copy-paste and use without the additional tool installation step be available.

Megh, I'll try to put one together and time-box to about a half-hour. In the worst case, I think this method is something support folks can do manually.

Reply:
@meghfossa I've added a script if you want to look it over. For now, I think I won't mention it in the docs. I'd rather a support person be guiding them so that they can have more context.